extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetInstructions.java

```diff
@@ -471,7 +473,7 @@ private ReadOnly(
             this.baseNameForPartitionedParquetData = baseNameForPartitionedParquetData;
             this.fileLayout = fileLayout;
             this.tableDefinition = tableDefinition;
-            this.indexColumns = indexColumns;
+            this.indexColumns = indexColumns == null ? null : Collections.unmodifiableCollection(indexColumns);
```

Comment:
You need a deeper copy, unless you can prove (to yourself) that all invocations of this private constructor are safe/copies.